internal/agentdiscovery/agentdiscovery.go

```diff
@@ -0,0 +1,159 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022 Datadog, Inc.
+
+// Package agentdiscovery provides a mechanism for automatically discovering a
+// Datadog agent and its available features
+package agentdiscovery
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"time"
+)
+
+const (
+	// defaultSocketAPM specifies the socket path to use for connecting to the trace-agent.
+	defaultSocketAPM   = "/var/run/datadog/apm.socket"
+	defaultHostname    = "localhost"
+	defaultPort        = "8126"
+	defaultHTTPTimeout = 2 * time.Second
+)
+
+// Features is functionality reported as available by the trace-agent
+type Features struct {
+	// Endpoints are the available data ingestion endpoints such as /traces,
+	// /profiles, etc.
+	Endpoints []string `json:"endpoints"`
+	// ClientDropP0s indicates whether the tracer client should [ ... SOMETHING ... ]
+	ClientDropP0s bool `json:"client_drop_p0s"`
+	// StatsdPort is the UDP port where the agent accepts DogStatsD metrics
+	StatsdPort int `json:"statsd_port"`
+	// FeatureFlags are enabled agent feature flags
+	FeatureFlags []string `json:"feature_flags"`
+}
+
+// AgentFeatures queries the agent's /info endpoint
+func AgentFeatures(addr string, client *http.Client) (*Features, error) {
```

Comment:
nit:
Is there a way we can make this name not stutter when used (i.e. not `agent.AgentFeatures()` )


`agent.Features()` would be ideal, but there is a `Features` type already, which is appropriate.


Maybe `LoadFeatures`? `QueryFeatures`?

Reply:
`LoadFeatures` sounds good to me